From 75819a49396593d14a766176bc51953d250be409 Mon Sep 17 00:00:00 2001
From: Ilya Nikolaevskiy <ilnik@chromium.org>
Date: Fri, 10 Jan 2025 01:18:31 -0800
Subject: [PATCH] Don't call AsyncMappingIsNonBlocking for nonmappable frames in trace log

It's forbidden to call AsyncMappingIsNonBlocking on non-HasMappableGpuBuffer frames.

This fixes a crash caused by a CHECK if the incoming frame isn't backed by GMB and disabled_by_default-webrtc category is used in tracing.

Bug: 388796101
Change-Id: I6328426a14bec8fbaa3a662ddc379e23a1461f72
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/6163627
Reviewed-by: Henrik Boström <hbos@chromium.org>
Reviewed-by: Markus Handell <handellm@google.com>
Commit-Queue: Ilya Nikolaevskiy <ilnik@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1404657}
---

--- a/third_party/blink/renderer/platform/peerconnection/webrtc_video_track_source.cc
+++ b/third_party/blink/renderer/platform/peerconnection/webrtc_video_track_source.cc
@@ -215,7 +215,8 @@
               adapter_resources_->GetFeedback().require_mapped_frame,
               "HasMappableGmb", current_frame->HasMappableGpuBuffer(),
               "AsyncMappingIsNonBlocking",
-              current_frame->AsyncMappingIsNonBlocking());
+              current_frame->HasMappableGpuBuffer() &&
+                  current_frame->AsyncMappingIsNonBlocking());
   // Map the GMB here if we know that the mapped image is required downstream.
   // If the feedback has reached the capturer, this is a no-op as the frame is
   // premapped. Otherwise it moves the mapping out of the encode operation,
